### ISSUE TITLE
Feature Request: Use only MPU401 to COM functionality w/o Sound Blaster emulation

### ISSUE THREAD

**Issue by hjnijlunsing** [crazii/SBEMU#93]
Scenario:
- I would like to use an ESS-Solo-1 (with existing Sound Blaster support) in combination with the MPU401 Com port ability, so I can play Doom with MIDI on my external unit.
- SoftMPU is not an option for this since Doom is a protected mode game

**Comment by crazii** [comment]
It can be done. I think @jiyunomegami is more familiar with that, but I don't know if he has time do to it.

**Comment by jiyunomegami** [comment]
Can you use something like sbemu /MCOM1 /A240 to set the SBEMU sound blaster address to 240, and use address 220 with the ESS card?
You may need to correct the BLASTER variable after running SBEMU.
And you would need another sound card or onboard ICH/HDA that is supported by SBEMU. FM won't work now because the address is hardcoded at 388.

If that doesn't work then it may be necessary to add an option to disable SB/FM port trapping. Does the Solo-1 need a TSR?

**Comment by hjnijlunsing** [comment]
Thanks for the feedback!

Unfortunately it does not work: with the JEMMEX/JLOAD combination my ESS-Solo does not produce any sound anymore.
This is even before running SBEMU; the Solo-1 does require a small TSR.

So the initial feature request will also not work

**Comment by jiyunomegami** [comment]
Do you have hdpmi32i running?
If so, it might not be the fault of jemmex/qpiemu.
Try running hdpmi32i -d to disable it.

I do have the Linux driver for the ESS Allegro-1 (ES1988/1989) working in my fork.
It shouldn't be difficult to port the Linux Solo-1 driver too, but I don't have a card to test with.

**Comment by hjnijlunsing** [comment]
You're right hdpmi32i was the culprit; however after disabling SBEMU fails to load of course ;-)

**Comment by crazii** [comment]
> I do have the Linux driver for the ESS Allegro-1 (ES1988/1989) working in my fork.

Is it possible to merge your fork to mainstream? I plan to release the beta4 (with some buggy changes maybe) and branch the current code and make some aggressive changes on the DPMI part.

**Comment by crazii** [comment]
@jiyunomegami BTW what's your id on VOGONS? I'll add credits on the next beta release. :)
and I guess the id of @hjnijlunsing is [veelstekel](https://www.vogons.org/memberlist.php?mode=viewprofile&u=51717) right?

**Comment by hjnijlunsing** [comment]
> I guess the id of @hjnijlunsing is [veelstekel](https://www.vogons.org/memberlist.php?mode=viewprofile&u=51717) right?
That is correct!

**Comment by jiyunomegami** [comment]
> > I do have the Linux driver for the ESS Allegro-1 (ES1988/1989) working in my fork.
> 
> Is it possible to merge your fork to mainstream? I plan to release the beta4 (with some buggy changes maybe) and branch the current code and make some aggressive changes on the DPMI part.

The ESS Allegro-1 driver along with the other Linux drivers all use the Linux headers/sound driver source code. If that is OK I can make a pull request over the weekend.

My ID on VOGONS is the same as my github ID, jiyunomegami

**Comment by crazii** [comment]
> > > I do have the Linux driver for the ESS Allegro-1 (ES1988/1989) working in my fork.
> > 
> > 
> > Is it possible to merge your fork to mainstream? I plan to release the beta4 (with some buggy changes maybe) and branch the current code and make some aggressive changes on the DPMI part.
> 
> The ESS Allegro-1 driver along with the other Linux drivers all use the Linux headers/sound driver source code. If that is OK I can make a pull request over the weekend.
> 
> My ID on VOGONS is the same as my github ID, jiyunomegami

Yes, please, they can be added as long as there's no major conflicts or potential problems that cause the program unstable. we can figure out a better solution to make it more maintainable later.
